ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -638,10 +638,6 @@ public void recoveryFailureTest() {
     info = new MessageInfo(state.deletedKeys.iterator().next(), CuratedLogIndexState.PUT_RECORD_SIZE,
         Utils.getRandomShort(TestUtils.RANDOM), Utils.getRandomShort(TestUtils.RANDOM), Utils.Infinite_Time);
     doRecoveryFailureTest(info, StoreErrorCodes.Initialization_Error);
-    // recovery info contains a Ttl Update for a key that does not exist
-    info = new MessageInfo(state.getUniqueId(), CuratedLogIndexState.TTL_UPDATE_RECORD_SIZE, false, true,
-        Utils.getRandomShort(TestUtils.RANDOM), Utils.getRandomShort(TestUtils.RANDOM), state.time.milliseconds());
-    doRecoveryFailureTest(info, StoreErrorCodes.ID_Not_Found);
```

Comment:
Why this case returns `Index_Creation_Failure` ? According to your comment to @pnarayanan  , this shouldn't return any exception.

Reply:
im missing something here - this is removed code and don't think there is a case that returns `Index_Creation_Failure`. Did you import these changes locally and run and see some tests failing?

Reply:
I _may_ have force pushed on the first commit at some point. Maybe you are reviewing an older diff?

Reply:
Yes, locally run got `Index_Creation_Failure`. Same to the latest diff.

Reply:
My base code is with your latest change and then bring 641 - 644 back, it returns `Index_Creation_Failure`.